src/HealthChecks.CassandraDb/CassandraDbHealthCheck.cs

```diff
@@ -0,0 +1,43 @@
+using Cassandra;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace HealthChecks.CassandraDb;
+
+public class CassandraDbHealthCheck : IHealthCheck
+{
+    private readonly CassandraDbOptions _options;
+
+    public CassandraDbHealthCheck(CassandraDbOptions options)
+    {
+        _options = Guard.ThrowIfNull(options);
+
+        if (_options.ContactPoint is null && _options.Keyspace is null)
+        {
+            throw new ArgumentException("A connection or connection string must be set!", nameof(options));
+        }
+    }
+
+    public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+    {
+        try
+        {
+            var builder = Cluster.Builder().AddContactPoint(_options.ContactPoint);
```

Comment:
With the current approach, we would build a new instance every time the health check would be invoked. Which I expect would be bad for performance and could even lead to some resource leaks (nothing is getting disposed here).

We have changed many of our health checks similar to what I described in #2040: for all the client instances that are thread-safe and should be used as singleton, we don't create our own instances, but expect it to be resolved from the DI (so the user registers a singleton and we just resolve).

I've done some quick web search, but have failed to find dependency injection best practices for working with the C# client of CassandarDb.

Should we just expect the users to inject the cluster into the DI and solve it?

For example this is what we do for MongoDB:

We allow the users to specify a `Func<IServiceProvider, IMongoClient>` factory:

https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/blob/36bf1712ff59416b05a84f4b900ed3422ba39fd6/src/HealthChecks.MongoDb/DependencyInjection/MongoDbHealthCheckBuilderExtensions.cs#L33

when it's not specified, we just try to resolve it from the DI container:

https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/blob/36bf1712ff59416b05a84f4b900ed3422ba39fd6/src/HealthChecks.MongoDb/DependencyInjection/MongoDbHealthCheckBuilderExtensions.cs#L51